test/api-documentation.test.js

```diff
@@ -571,4 +589,237 @@ describe('<api-documentation>', function() {
+              'method',
+              html`
+                <anypoint-item slot="custom-base-uri" value="http://customServer.com">
+                  Server 1 - http://customServer.com
+                </anypoint-item>
+                <anypoint-item slot="custom-base-uri" value="http://customServer.com/{version}">
+                  Server 2 - http://customServer.com/{version}
+                </anypoint-item>
+              `
+            );
+
+            await nextFrame();
+          });
+
+          it('should load servers', () => {
+            assert.lengthOf(element.servers, 1);
+          });
+
+          it('should set serversCount', () => {
+            assert.equal(element.serversCount, 4);
+          });
+
+          it('should update selectedServerValue and selectedServerType using the first available server', () => {
+            assert.equal(element.selectedServerValue, 'http://{instance}.domain.com/');
+            assert.equal(element.selectedServerType, 'server');
+          });
+
+          it('should not change the baseUri property', () => {
+            assert.isUndefined(element.baseUri);
+          });
+
+          it('should render api-server-selector', () => {
+            assert.exists(element.shadowRoot.querySelector('api-server-selector'));
+          });
+
+          it('should not hide api-server-selector', () => {
+            assert.isFalse(element.shadowRoot.querySelector('api-server-selector').hidden);
+          });
+
+          describe('when navigating to something other than a method or an endpoint', () => {
+            beforeEach(async () => {
+              element.selectedType = 'summary';
+              element.selected = 'summary';
+
+              await aTimeout();
+            });
+
+            it('should hide api-server-selector', () => {
+              assert.isTrue(element.shadowRoot.querySelector('api-server-selector').hidden);
+            });
+          });
+
+          describe('when serverCount changes to less than 2', () => {
+            let serverSelector;
+
+            beforeEach(async () => {
+              serverSelector = element.shadowRoot.querySelector('api-server-selector');
+
+              serverSelector.dispatchEvent(new CustomEvent('servers-count-changed', { detail: { serversCount: 1 } }));
+
+              await aTimeout();
+            });
+
+            it('should set serversCount', () => {
+              assert.equal(element.serversCount, 1);
+            });
+
+            it('should hide api-server-selector', () => {
+              assert.isTrue(serverSelector.hidden);
+            });
+          });
+
+          describe('when selecting a slot server', () => {
+            beforeEach(() => {
+              const event = {
+                detail: {
+                  selectedValue: 'http://customServer.com',
+                  selectedType: 'slot'
+                }
+              };
+              element.dispatchEvent(new CustomEvent('api-server-changed', event));
```

Comment:
This should work when this event is dispatched on `api-console` or `window` object, not only the element.